src/runmode-dpdk.c

```diff
@@ -1102,6 +1130,87 @@ static int DeviceConfigureIPS(DPDKIfaceConfig *iconf)
     SCReturnInt(0);
 }
 
+static void DeviceRingNameInit(char *name_out, const char *format, uint16_t q_num)
+{
+    char name[RTE_RING_NAMESIZE];
+    char q_num_str[5];
+    char *q_pos;
+
+    if (q_num > 9999)
+        FatalError(SC_ERR_DPDK_CONF, "Number of queues must not exceed 10000 queues per interface");
+    q_pos = strstr(format, DPDK_CONFIG_DEFAULT_QUEUE_NUM_SPECIFIER);
+    snprintf(name, q_pos - format + 1, "%s", format);
+    snprintf(q_num_str, sizeof(q_num_str), "%" PRIu16, q_num);
+    strlcat(name, q_num_str, strlen(q_num_str));
+    // copy the rest after queue number specifier
+    strlcat(name, q_pos + strlen(DPDK_CONFIG_DEFAULT_QUEUE_NUM_SPECIFIER),
+            sizeof(name) - strlen(name) - 1);
+    strlcpy(name_out, name, sizeof(name));
```

Comment:
the size arg needs to be the size of the target buffer, which should be passed as an arg to this function

Reply:
Argument has been added to the function.